third_party/move/move-vm/profiler/src/probe.rs

```diff
@@ -0,0 +1,90 @@
+//! A profiler that emits USDT probes.
+
+use move_vm_types::instr::Instruction;
+use once_cell::sync::Lazy;
+use std::time::Instant;
+
+#[usdt::provider]
+mod vm_profiler {
+    fn function_entry(function: &str) {}
+    fn function_exit(function: &str, nanos: u64) {}
+
+    fn instruction_entry(instruction: &str) {}
+    fn instruction_exit(instruction: &str, nanos: u64) {}
+}
```

Comment:
Currently I only support entry/exit probes for function and instruction execution.

We might want more probes in the future, but this can be easily extended.

Reply:
I think we discussed this offline, but the instruction probes don't do anything unless they're enabled in the DTrace script, so there is no harm to leaving this in.